cranelift/codegen/src/verifier/mod.rs

```diff
@@ -1322,22 +1319,7 @@ impl<'a> Verifier<'a> {
                 let args_else = block_else.args_slice(&self.func.dfg.value_lists);
                 self.typecheck_variable_args_iterator(inst, iter, args_else, errors)?;
             }
-            ir::InstructionData::BranchTable {
-                table,
-                destination: block,
-                ..
-            } => {
-                let arg_count = self.func.dfg.num_block_params(*block);
-                if arg_count != 0 {
-                    return errors.nonfatal((
-                        inst,
-                        self.context(inst),
-                        format!(
-                            "takes no arguments, but had target {} with {} arguments",
-                            block, arg_count,
-                        ),
-                    ));
-                }
+            ir::InstructionData::BranchTable { table, .. } => {
                 for block in self.func.stencil.dfg.jump_tables[*table].iter() {
```

Comment:
Iter skips the default block, right?

Reply:
`iter` includes the default block. I decided that `as_slice`, `as_slice_mut`, `iter`, and `iter_mut` would iterate the whole vector, and that access to just the table or default block would be through `table_slice` and `default_block`.

Reply:
Perhaps it would be better to go the other way, and introduce a special function for accessing a slice for the whole vector, keeping the existing functions as excluding the default?

Reply:
Another option would be to not use the regular names for conversion to a slice or iterator at all, but instead explicitly call out which part is returned for each function, both those returning everything and those returning only a part.

Reply:
I think I'm going to rework this to keep the existing behavior consistent, and introduce new functions to iterate the whole range. Thanks for pointing this out :)